### PR TITLE
[Mini] Read species distribution from OPMD - part 4

### PR DESCRIPTION
**Pull request by LDAmorim**
Hi all,

This small PR follows #883, #859 and #847 as it is the forth step towards implementing the option of reading/initializing a species from an external openPMD file that contains a realistic sample of N total particles, with mass, charge (q_p), position, and momentum information.

For this PR what I changed was:

* **Source/Particles/PhysicalParticleContainer.cpp** - to avoid warnings/errors from overwriting the variable `i` for openPMD iteration and the for cycle that iterates the particles. And to correct the particle weighing.

Together with @ax3l and @MaxThevenet we are testing the issue #952 with this branch and will update it once we fix it.

When running an example with only 8 particles in 1 node with:

[e-_primary_s1_drift_opmd.h5.zip](https://github.com/ECP-WarpX/WarpX/files/4541268/e-_primary_s1_drift_opmd.h5.zip)
[input_script.txt](https://github.com/ECP-WarpX/WarpX/files/4541275/input_script.txt)
I get the correct output:
[output.txt](https://github.com/ECP-WarpX/WarpX/files/4541270/output.txt)comp_codes_vx
comp_codes_x
comp_codes_y
comp_codes_vy
comp_codes_z
comp_codes_vz

```python
Weight of macroparticles =  780188705.8900522  from openPMD
In WarpX weight =  [-7.80188706e+08 -7.80188706e+08 -7.80188706e+08 -7.80188706e+08
 -7.80188706e+08 -7.80188706e+08 -7.80188706e+08 -7.80188706e+08]
```

Thanks,
Diana

### CODE REVIEW

File: Source/Particles/PhysicalParticleContainer.cpp [hunk 1]
```diff
@@ -382,10 +382,10 @@ PhysicalParticleContainer::AddPlasmaFromFile(const std::string s_f,
 
         amrex::Real weight;
         if (q_tot != 0.0){
-            weight = q_tot/(p_q*amrex::Real(npart));
+            weight = abs(q_tot)/(abs(charge)*amrex::ParticleReal(npart));
```
Comment (ax3l) [inline Source/Particles/PhysicalParticleContainer.cpp:385]: This somehow picks up an AMReX int function.
Instead include
```C++
#include <cmath>
```
and
```suggestion
            weight = std::abs(q_tot)/(std::abs(charge)*amrex::ParticleReal(npart));
```

File: Source/Particles/PhysicalParticleContainer.cpp [hunk 2]
```diff
@@ -382,10 +382,10 @@ PhysicalParticleContainer::AddPlasmaFromFile(const std::string s_f,
...
         }
         else {
-            weight = charge;
+            weight = abs(p_q); //TODO: Explain how codes can make use of this option
```
Comment (ax3l) [inline Source/Particles/PhysicalParticleContainer.cpp:388]:
```suggestion
            weight = std::abs(p_q); //TODO: Explain how codes can make use of this option
```

**Comment by LDAmorim** [comment]
When I ran that input with that external file, but with 2 MPI ranks I get:
An empty `Backtrace.1.2` file.
[Backtrace.1.0.txt](https://github.com/ECP-WarpX/WarpX/files/4541305/Backtrace.1.0.txt)
[output.txt](https://github.com/ECP-WarpX/WarpX/files/4541309/output.txt)
[error.txt](https://github.com/ECP-WarpX/WarpX/files/4541307/error.txt)

**Comment by ax3l** [comment]
Dang, that is the same  #939

**Comment by ax3l** [comment]
I think I might have accidentally include these in #956 :sweat:

File: Source/Particles/PhysicalParticleContainer.cpp [hunk 3]
```diff
@@ -382,10 +381,10 @@ PhysicalParticleContainer::AddPlasmaFromFile(const std::string s_f,
 
         amrex::Real weight;
         if (q_tot != 0.0){
-            weight = q_tot/(p_q*amrex::Real(npart));
+            weight = std::abs(q_tot)/(std::abs(charge)*amrex::ParticleReal(npart));
```
Comment (MaxThevenet) [inline Source/Particles/PhysicalParticleContainer.cpp:384]:
```suggestion
            weight = std::abs(q_tot/charge)/amrex::ParticleReal(npart);
```

**Review by MaxThevenet — COMMENTED** [review]
Thanks for this PR! 
Looks good to me, just see minor comment below.

File: Source/Particles/PhysicalParticleContainer.cpp (hunk 3 above)
Comment (LDAmorim) [reply Source/Particles/PhysicalParticleContainer.cpp:384]: That's a good point @MaxThevenet !

**Comment by MaxThevenet** [comment]
These changes are included in #956, therefore I will close this PR.

**Pull request closed without merging**